eolearn/visualization/eoexecutor.py

```diff
@@ -97,40 +100,49 @@ def _create_dependency_graph(self) -> str:
         dot = self.eoexecutor.workflow.dependency_graph()
         return base64.b64encode(dot.pipe()).decode()
 
-    def _get_exception_stats(self) -> list[tuple[str, str, list[tuple[str, int]]]]:
-        """Creates aggregated stats about exceptions"""
-        formatter = HtmlFormatter()
-        lexer = pygments.lexers.get_lexer_by_name("python", stripall=True)
+    def _get_exception_stats(self) -> list[tuple[str, str, list[_ErrorSummary]]]:
+        """Creates aggregated stats about exceptions
+
+        Returns tuples of form (name, uid, [error_summary])
+        """
 
-        exception_stats: defaultdict[str, defaultdict[str, int]] = defaultdict(lambda: defaultdict(lambda: 0))
+        exception_stats: defaultdict[str, dict[str, _ErrorSummary]] = defaultdict(dict)
 
-        for workflow_results in self.eoexecutor.execution_results:
-            if not workflow_results.error_node_uid:
+        for execution, results, execution_idx in zip(
+            self.eoexecutor.execution_names, self.eoexecutor.execution_results, it.count()
+        ):
```

Comment:
You could also do 

```suggestion
        for execution_idx, (execution, results) in enumerate(zip(
            self.eoexecutor.execution_names, self.eoexecutor.execution_results)):
```

and loose the extra import

Reply:
at some point this could fit into one line, but now we can remove it i guess, gotta see how the lines break